qcodes/instrument/parameter.py

```diff
@@ -282,29 +282,33 @@ def __init__(self, name: str,
         self.get_parser = get_parser
         self.set_parser = set_parser
 
-        # record of latest value and when it was set or measured
-        # what exactly this means is different for different subclasses
-        # but they all use the same attributes so snapshot is consistent.
-        self._latest: Dict[str, Optional[Union[ParamDataType,
-                                               ParamRawDataType,
-                                               datetime]]]
-        self._latest = {'value': None, 'ts': None, 'raw_value': None}
+        # ``_Cache`` stores "latest" value (and raw value) and timestamp
+        # when it was set or measured
+        self.cache = _Cache(self, max_val_age=max_val_age)
+        # ``GetLatest`` is left from previous versions where it would
+        # implement a subset of features which ``_Cache`` has.
+        # It is left for now for backwards compatibility reasons and shall
+        # be deprecated and removed in the future versions.
         self.get_latest: GetLatest
-        self.get_latest = GetLatest(self, max_val_age=max_val_age)
+        self.get_latest = GetLatest(self)
 
         self.get: Callable[..., ParamDataType]
-        if hasattr(self, 'get_raw') and not getattr(self.get_raw, '__qcodes_is_abstract_method__', False):
+        get_raw_is_abstract = getattr(self.get_raw,
+                                      '__qcodes_is_abstract_method__', False)
```

Comment:
Is this refactor safe? This will try to look up attributes on get_raw before checking that there is a get_raw method

Reply:
oh, you're right. good catch, i'll "revert" this.

Reply:
done.